lib/stats/stats-counter.h

```diff
@@ -29,45 +29,56 @@
 
 typedef struct _StatsCounterItem
 {
-  atomic_gssize value;
+  union
+  {
+    atomic_gssize value;
+    atomic_gssize *value_ref;
+  };
   gchar *name;
   gint type;
+  gboolean external;
 } StatsCounterItem;
 
 
+static gboolean
+stats_counter_read_only(StatsCounterItem *counter)
+{
+  return counter->external;
+}
+
 static inline void
 stats_counter_add(StatsCounterItem *counter, gssize add)
 {
-  if (counter)
+  if (counter && !stats_counter_read_only(counter))
```

Comment:
I would use an assert here and ensure in higher layers that we are not calling counter changing functions in case it is a read-only counter.